eclass/gradle.eclass

```diff
@@ -0,0 +1,208 @@
+# Copyright 2021-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: gradle.eclass
+# @MAINTAINER:
+# Gentoo Java Project <java@gentoo.org>
+# @AUTHOR:
+# Florian Schmaus <flow@gentoo.org>
+# @BLURB: common ebuild functions for gradle-based packages.
+# @DESCRIPTION:
+# This eclass provides support for the gradle build system.  There
+# are currently two approaches to using gradle in ebuilds.  You can either
+# depend on a gradle system-wide installation from a gradle ebuild, typically
+# dev-java/gradle-bin, or, bundle gradle with the ebuild.
+#
+# To use a system-wide gradle installation, set EGRADLE_MIN and
+# EGRADLE_MAX_EXCLUSIVE and declare a BDEPEND on the gradle package.
+# @CODE
+# inherit gradle
+# EGRADLE_MIN=7.3
+# EGRADLE_MAX_EXCLUSIVE=8
+#
+# BDEPEND="|| (dev-java/gradle-bin:7.3 dev-java/gradle-bin:7.4)
```

Comment:
```suggestion
# BDEPEND="|| (dev-java/gradle-bin:7.3 dev-java/gradle-bin:7.4 )"
```

but I'm a bit surprised that the eclass doesn't give me a depstring given we tell it all of the information?

Reply:
I wonder if we can do it based on =dev-java/gradle-bin-${EGRADLE_MIN}* (iterate until EGRADLE_MAX_EXCLUSIVE), i.e. skipping slots but using * deps?